app/services/provider_interface/active_application_statuses_by_provider.rb

```diff
@@ -0,0 +1,42 @@
+module ProviderInterface
+  class ActiveApplicationStatusesByProvider
+    attr_reader :provider
+
+    def initialize(provider)
+      @provider = provider
+    end
+
+    def call
+      grouped_course_data.map do |course_data|
+        courses = course_data.last
```

Comment:
We seem to be disregarding the `group_by(&:id)` from `grouped_course_data` here, is it necessary?

Reply:
Yes. We need it in order to correctly construct each row of data. An example of what the report will look like can be found here: https://trello.com/c/JmMsavpG/4134-construct-status-of-active-applications-report